multiqc/modules/picard/ValidateSamFile.py

```diff
@@ -0,0 +1,369 @@
+                problem_type, name = values[0].split(':')
+                yield problem_type, name, int(values[1])
+            elif l.startswith('Error Type'):
+                header_passed = True
+
+
+def _add_section_to_report(self, data):
+    """
+        Adds found data to the report via several HTML generators
+    """
+
+    # Count samples that have errors and/or warnings
+    pass_count = error_count = only_warning_count = 0
+    for _, sample_data in data.items():
+        if sample_data['file_validation_status'] == 'fail':
+            error_count += 1
+        elif sample_data['file_validation_status'] == 'warn':
+            only_warning_count += 1
+        else:
+            pass_count += 1
+
+    # Add overview note
+    plot_html = []
+    note_html = _generate_overview_note(
+                        pass_count=pass_count,
+                        only_warning_count=only_warning_count,
+                        error_count=error_count,
+                        total_count=len(data)
+                    )
+    plot_html.append(note_html)
+
+    # Add the detailed table, but only if we have something to show
+    if error_count or only_warning_count:
+        table_html = _generate_detailed_table(data)
+        plot_html.append(table_html)
+
+    # Finally, add the html to the report as a section
+    self.add_section(
+            name='SAM/BAM File Validation',
+            anchor='picard_validatesamfile',
+            description='This tool reports on the validity of a SAM or BAM file relative to the SAM-format specification.',
+            helptext='''
+            A detailed table is only shown if a errors or warnings are found. Details about the errors and warnings are only shown if a SUMMARY report was parsed.
+
+            For more information on the warnings, errors and possible fixes please read [this broadinstitute article](https://software.broadinstitute.org/gatk/documentation/article.php?id=7571).''',
+            plot="\n".join(plot_html),
+    )
+
+
+def _add_data_to_general_stats(self, data):
+    """
+        Add data for the general stats in a Picard-module specific manner
+    """
+    headers = _get_general_stats_headers()
+
+    self.general_stats_headers.update(headers)
+
+    general_data = dict()
+    for sample in data:
+        general_data[sample] = {key: data[sample][key] for key in ('ERROR_count', 'WARNING_count', 'file_validation_status')}
+
+        if sample not in self.general_stats_data:
+                self.general_stats_data[sample] = dict()
+        self.general_stats_data[sample].update(general_data[sample])
+
+    self.general_stats_addcols(data=general_data, headers=headers)
+
+
+def _get_general_stats_headers():
+    """
+        Returns a header dict for the general stats collumns
+    """
+    headers = OrderedDict()
+
+    headers['file_validation_status'] = {
+        'title': "File-Validation Status",
+        'description': 'pass/warn/fail',
+        'hidden': False,
```

Comment:
This is the default value, so not needed. Though it could be nice to have `True` if all values `pass`, hiding the column by default if there's nothing interesting to show.